src/js/components/Tag/Tag.js

```diff
@@ -11,23 +11,27 @@ import { Text } from '../Text';
 
 import { StyledRemoveButton, StyledTagButton } from './StyledTag';
 
-const Tag = forwardRef(({ name, value, onRemove, onClick, ...rest }, ref) => {
+const Tag = forwardRef(
+  ({ name, value, size, onRemove, onClick, ...rest }, ref) => {
   const theme = useContext(ThemeContext) || defaultProps.theme;
 
   const containerProps = {
     ref,
     align: 'center',
     background: theme.tag.background,
     border: theme.tag.border,
-    round: theme.tag.round,
+    round: theme.tag.size?.[size]?.round || theme.tag.round,
     ...rest,
   };
   const contents = (
-    <Box width={{ min: 'min-content' }} pad={theme.tag.pad}>
-      <Text>
-        {name && <Text {...theme.tag.name}>{name}</Text>}
-        {name && value ? theme.tag.separator : ''}
-        {value && <Text {...theme.tag.value}>{value}</Text>}
+    <Box
+      width={{ min: 'min-content' }}
+      pad={theme.tag.size?.[size]?.pad || theme.tag.pad}
+    >
+      <Text size={size}>
```

Comment:
Why do we need the size both here and in the contained Text elements? I would think one or the other would suffice. Maybe just keeping this one?

Reply:
Without it here you get a span with a line-height of medium which makes it bigger than the contained text